src/monster.cpp

```diff
@@ -1957,12 +1957,16 @@ void Monster::changeHealth(int32_t healthChange, bool sendHealthChange/* = true*
 	Creature::changeHealth(healthChange, sendHealthChange);
 }
 
-bool Monster::challengeCreature(Creature* creature)
+bool Monster::challengeCreature(Creature* creature, bool force/* = false*/)
 {
 	if (isSummon()) {
 		return false;
 	}
 
+	if (!mType->info.isChallengeable && !force) {
```

Comment:
Here `force` doesn't really work as its name implies, regardless of whether `force` is `true` or `false`, it will not change any behavior

`force` is supposed to be a parameter that forces the creature regardless of whether it is challengeable or no
![image](https://user-images.githubusercontent.com/28090948/130485763-95d9f468-d0cd-4cef-b1c2-ce84be84ef7f.png)

Reply:
True, thanks for spotting 🙉

Reply:
Actually, it works as expected 🤔 

if monster **is not** challengeable **AND** call **is not** forced, then and only then **return false**.

So, examples:

| Challengeable | Force | Will exit early? |
|---------------|-------|----------------|
| Y                      | N        | N                       |
| Y                      | Y        | N                       |
| N                      | N        | Y                       |
| N                      | Y        | N                       |